### PR TITLE
Added ISBN integrity checker

### PR DESCRIPTION
**Pull request by oscargus**
ISBN integrity checker

- [x] Change in CHANGELOG.md described
- [x] Tests created for changes

### CODE REVIEW

File: src/main/java/net/sf/jabref/logic/integrity/ISBNChecker.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+    @Override
+    public List<IntegrityMessage> check(BibEntry entry) {
+        if (!entry.hasField("isbn")) {
+            return Collections.emptyList();
+        }
+
+        // Check that the ISSN is on the correct form
+        String isbn = entry.getFieldOptional("isbn").get().trim().replace("-", "");
+        Matcher isbnMatcher = ISBN_PATTERN.matcher(isbn);
```
Comment (tobiasdiez) [inline src/main/java/net/sf/jabref/logic/integrity/ISBNChecker.java]: See https://github.com/JabRef/jabref/pull/1590#discussion_r70992658

Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/integrity/ISBNChecker.java]: Again, how to handle invalid data? Here it is slightly more obvious as the dashes are ignored, but should one store both version? Some people may prefer the dashes in particular positions...

I do not see this happening in this PR.

Comment (simonharrer) [inline src/main/java/net/sf/jabref/logic/integrity/ISBNChecker.java]: I would simply create an ISBN class in the model which receives a String and then can verify whether this String is a valid ISBN or not. Hence, isolate the check logic. I would not care about storing here.

**Comment by simonharrer** [comment]
LGTM just extract the validation logic, as it could be useful elsewhere, and then merge this in.

**Comment by oscargus** [comment]
OK! Without the `toString()` it makes sense (and is easier to cope with than ISSN, as there isn't an "almost correct" ISBN format...).

**Pull request merged**